src/adera/Machines/Rocket.cpp

```diff
@@ -125,7 +124,8 @@ void SysMachineRocket::update_physics(ActiveScene& rScene)
         bool fail = false;
         for (auto& resource : machine.m_resourceLines)
         {
-            auto const* src = m_scene.reg_try_get<MachineContainer>(resource.m_sourceEnt);
+            auto const* pipe = resource.m_source.get_if<wiretype::Pipe>();
+            auto const* src = rScene.reg_try_get<MachineContainer>(pipe->m_source);
             if (!src)
             {
                 std::cout << "Error: no source found\n";
```

Comment:
Check not only for the presence of resources, but actually ensure that enough resources are available (in accordance with the consumption ratios) before allowing physics update

-- Capital Asterisk, somewhere I can't find